### PR TITLE
Added the ability to check default KubeletConfig

### PR DESCRIPTION
**Pull request by Vincent056**
Compliance Operator will now fetch `KubeletConfig` for each node, and compare and save only consistent `KubeletConfig` under `/kubeletconfig/role/{{role}}`

### CODE REVIEW

**Comment by Vincent056** [comment]
Needed to add the e2e test later.

File: cmd/manager/scap.go
Comment (jhrozek) [inline cmd/manager/scap.go:69]: Wouldn't this be more go-like to use a map of string-to a slice of strings?

File: cmd/manager/scap.go
Comment (jhrozek) [inline cmd/manager/scap.go:201]: Can we collapse these two into one?

File: cmd/manager/scap.go
Comment (Vincent056) [reply cmd/manager/scap.go:69]: thanks for the advice, just fixed that

**Comment by Vincent056** [comment]
e2e and unit test has been added

File: cmd/manager/scap.go
Comment (jhrozek) [inline cmd/manager/scap.go:281]: this would read much cleaner with named return parameters

File: cmd/manager/scap.go
Comment (jhrozek) [inline cmd/manager/scap.go:287]: can you add an example of the dump path? Otherwise it's not easy to see what the magic constants are

File: cmd/manager/scap.go
Comment (jhrozek) [inline cmd/manager/scap.go:599]: lower-case the local variable, please

File: cmd/manager/scap_test.go
Comment (jhrozek) [inline cmd/manager/scap_test.go:637]: extra blank line

File: pkg/utils/parse_arf_result.go
Comment (jhrozek) [inline pkg/utils/parse_arf_result.go:140]: Since these functions are only used in tests, can we move them outside CO code and into the tests?

File: cmd/manager/scap.go
Comment (jhrozek) [inline cmd/manager/scap.go:605]: I like the idea of only saving intersection! This way if something is different and we care, then we find out because the attribute will not be there. Nice!

File: tests/e2e/e2e_test.go
Comment (Vincent056) [reply tests/e2e/e2e_test.go:3650]: I created that one, it uses a variable in the filepath, /kubeletconfig/{{role}}, I wanted to test it out

Comment (Vincent056) [reply tests/e2e/e2e_test.go:3650]:
```
template:
  name: yamlfile_value
  vars:
    ocp_data: "true"
    filepath: '/kubeletconfig/role'
    filepath_suffix: var_role
    yamlpath: ".tlsCipherSuites[:]"
    xccdf_variable: var_kubelet_tls_cipher_suites_regex
    regex_data: true
```

**Comment by Vincent056** [comment]
E2E test passed

**Comment by jhrozek** [comment]
/hold
until we figure out the impact on docs and QE

**Review by jhrozek — APPROVED** [review]
/lgtm
Just please let's discuss how is QE and docs going to handle the feature..I understand that this brings the CO part and we need the content part, so it's probably just about synchronizing and we can't expect a pre-merge testing for something that spans several projects.

**Comment by jhrozek** [comment]
@mkumku see this PR for px-approved

**Comment by Vincent056** [comment]
Hi @Wiharris, @xiaojiey Wondering if I could get doc-approval and qe-approval here. Thanks!

**Comment by xiaojiey** [comment]
@Vincent056 I rerun the test again with https://github.com/ComplianceAsCode/compliance-operator/pull/78 and https://github.com/ComplianceAsCode/content/pull/9370, now the remediation works for most of rules. However, the remediation still doesn’t work for rules upstream-ocp4-kubelet-configure-event-creation and upstream-ocp4-kubelet-eviction-thresholds-set-hard-imagefs-inodesfree. Could you please help to check again? The operatoration log is available at https://privatebin.corp.redhat.com/?b01ec1409c312a54#AQNNx9fWyK2UDa8241zXWBXHqDMNaxiU5dY1W9SQBE4E
and the detailed log is available at: https://drive.google.com/drive/folders/1AJwRr6DFe0l02J3YeORnEKA369yrkgim. Maybe I can create a pre-merge bug to track.

**Comment by xiaojiey** [comment]
> @Vincent056 I rerun the test again with #78 and [ComplianceAsCode/content#9370](https://github.com/ComplianceAsCode/content/pull/9370), now the remediation works for most of rules. However, the remediation still doesn’t work for rules upstream-ocp4-kubelet-configure-event-creation and upstream-ocp4-kubelet-eviction-thresholds-set-hard-imagefs-inodesfree. Could you please help to check again? The operatoration log is available at https://privatebin.corp.redhat.com/?b01ec1409c312a54#AQNNx9fWyK2UDa8241zXWBXHqDMNaxiU5dY1W9SQBE4E and the detailed log is available at: https://drive.google.com/drive/folders/1AJwRr6DFe0l02J3YeORnEKA369yrkgim. Maybe I can create a pre-merge bug to track.
One more question: does the rule upstream-ocp4-kubelet-enable-iptables-util-chains will be supported? Thanks.

File: pkg/utils/parse_arf_result.go
Comment (Wiharris) [inline pkg/utils/parse_arf_result.go:771]: LGTM! Just a couple of comments and suggestions left below: 

```suggestion
// This function will take original remediation content, and a list of all values found in the configMap
```

File: pkg/utils/parse_arf_result.go
Comment (Wiharris) [inline pkg/utils/parse_arf_result.go:772]:
```suggestion
// It will processed, substitute the value in remediation content and return processed Remediation content
```

File: pkg/utils/parse_arf_result.go
Comment (Wiharris) [inline pkg/utils/parse_arf_result.go:773]:
```suggestion
// The return will be Processed-Remediation Content, Value-Used List, Un-Set List, and err if possible```

File: vendor/github.com/tidwall/gjson/SYNTAX.md
Comment (Wiharris) [inline vendor/github.com/tidwall/gjson/SYNTAX.md:3]:
```suggestion
A GJSON Path is a text string syntax that describes a search pattern for quickly receiving values from a JSON payload.
```

File: vendor/github.com/tidwall/gjson/SYNTAX.md
Comment (Wiharris) [inline vendor/github.com/tidwall/gjson/SYNTAX.md:18]:
```suggestion
The definitive implementation is [github.com/tidwall/gjson](https://github.com/tidwall/gjson).  
```

File: vendor/github.com/tidwall/gjson/SYNTAX.md
Comment (Wiharris) [inline vendor/github.com/tidwall/gjson/SYNTAX.md:23]:
```suggestion
A GJSON Path is intended to be easily expressed as a series of components separated by a `.` character. 
```

File: vendor/github.com/tidwall/gjson/SYNTAX.md
Comment (Wiharris) [inline vendor/github.com/tidwall/gjson/SYNTAX.md:49]:
```suggestion
In many cases you'll just want to retrieve values by object name or array index.
```

File: vendor/github.com/wI2L/jsondiff/README.md
Comment (Wiharris) [inline vendor/github.com/wI2L/jsondiff/README.md:244]:
```suggestion
However, note that it includes one limitation. `copy` operations cannot be inverted, as they are ambiguous (the reverse of a `copy` is a `remove`, which could then become either an `add` or a `copy`). As such, using this option disable the generation of `copy` operations (if option `Factorize()` is used) and replace them with `add`, albeit potentially at the cost of increased patch size.
```

**Comment by Wiharris** [comment]
HI Vincent,

Docc approved. LGTM! Just left a few comments/suggestions. :)




Thanks,
William

File: vendor/github.com/tidwall/gjson/SYNTAX.md
Comment (Vincent056) [reply vendor/github.com/tidwall/gjson/SYNTAX.md:3]: This is in vendor, we don't manage the code here

File: vendor/github.com/tidwall/gjson/SYNTAX.md
Comment (Vincent056) [reply vendor/github.com/tidwall/gjson/SYNTAX.md:18]: same as above

File: cmd/manager/aggregator.go
Comment (rhmdnd) [inline cmd/manager/aggregator.go:8]: Was this just a formatting thing that ended up in this PR? Or was this necessary for this change?

File: vendor/github.com/tidwall/gjson/SYNTAX.md
Comment (Vincent056) [reply vendor/github.com/tidwall/gjson/SYNTAX.md:23]: same as above

File: vendor/github.com/tidwall/gjson/SYNTAX.md
Comment (Vincent056) [reply vendor/github.com/tidwall/gjson/SYNTAX.md:49]: same as above

File: vendor/github.com/wI2L/jsondiff/README.md
Comment (Vincent056) [reply vendor/github.com/wI2L/jsondiff/README.md:244]: same as above

**Comment by Vincent056** [comment]
@Wiharris Thanks for the review, I have addressed all the comments

File: cmd/manager/aggregator.go
Comment (Wiharris) [reply cmd/manager/aggregator.go:8]: Yep, this one was by accident. Sorry about that!

**Comment by xiaojiey** [comment]
Per https://github.com/ComplianceAsCode/compliance-operator/pull/78#issuecomment-1228170729, it should be content issue. Will approve this PR and track with https://github.com/ComplianceAsCode/content/pull/9370 instead.

File: cmd/manager/aggregator.go
Comment (Vincent056) [reply cmd/manager/aggregator.go:8]: I think this was from `make generate`

File: cmd/manager/scap_test.go
Comment (rhmdnd) [inline cmd/manager/scap_test.go:602]: nit: `expectedFiguredResources`?

File: cmd/manager/scap_test.go
Comment (rhmdnd) [inline cmd/manager/scap_test.go:669]: nit: consistency*

**Review by rhmdnd — COMMENTED** [review]
Looks good, just some minor comments and questions inline.

File: cmd/manager/scap.go
Comment (mrogers950) [reply cmd/manager/scap.go:287]: It looks like yes as long as getKubeletConfigResourcePath() is not changed without it.

**Comment by mrogers950** [comment]
LGTM.

**Comment by JAORMX** [comment]
What's missing here? Seems to be on hold still.

**Comment by Vincent056** [comment]
Rebased with Operator-SDK commit

**Comment by Vincent056** [comment]
could anyone to lgtm again, Thanks! @jhrozek @mrogers950

**Review by jhrozek — APPROVED** [review]
/lgtm

**Pull request merged**
